library/alloc/src/sync.rs

```diff
@@ -500,6 +500,62 @@ impl<T> Arc<T> {
             Ok(elem)
         }
     }
+
+    /// Returns the inner value, if the `Arc` has exactly one strong reference.
+    ///
+    /// Otherwise, [`None`] is returned and the `Arc` is dropped.
+    ///
+    /// This will succeed even if there are outstanding weak references.
+    ///
+    /// If `unwrap_or_drop` is called on every clone of this `Arc`,
+    /// it is guaranteed that exactly one of the calls returns the inner value.
+    /// The similar expression `Arc::try_unwrap(this).ok()` does not
+    /// offer this guarantee.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(unwrap_or_drop)]
+    ///
+    /// use std::sync::Arc;
+    ///
+    /// let x = Arc::new(3);
+    /// let y = x.clone();
```

Comment:
I find `Arc::clone(x)` is clearer because in some situations `T::clone` could be heavy while `Arc::clone` is not.

Reply:
Great idea! Do you have an opinion on those identifiers I chose in the following lines? I’m not totally happy with them since they’re very long, but short ones seemed potentially confusing.

Reply:
It would need to be `Arc::clone(&x)`. Do you still prefer this?

Reply:
Yep I do, forgot the & (and took a long time to find it on mobile), but my point still stands. 😃

The identifiers are fine, I prefer explicit over implicit and here they are used in a doc comment so clarity is even more important.